### PR TITLE
owner: ignore the invalid admin jobs

### LINKED ISSUE #541: abnormal behavior when resuming a stopped changefeed
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

- start two captures
- start a changefeed
- stop the changefeed via `curl -X POST -d "admin-job=3&cf-id=xxx" http://127.0.0.1:8301/capture/owner/admin`
- resume the changefeed (in fact resume should make no sense) via `curl -X POST -d "admin-job=2&cf-id=xxx" http://127.0.0.1:8301/capture/owner/admin`, send HTTP request to both owner capture and non-owner capture

2. What did you expect to see?

returns some expected error messages

3. What did you see instead?

non owner node
```
2020-05-08 11:53:48.360571 I | http: panic serving 127.0.0.1:33912: runtime error: invalid memory address or nil pointer dereference
goroutine 1489 [running]:
net/http.(*conn).serve.func1(0xc000354780)
        net/http/server.go:1767 +0x139
panic(0x1d9d8c0, 0x3af1740)
        runtime/panic.go:679 +0x1b2
sync.(*Mutex).Lock(...)
        sync/mutex.go:74
github.com/pingcap/ticdc/cdc.(*Owner).EnqueueJob(0x0, 0xc00056a012, 0x24, 0x2, 0xc002dfc4c0, 0x0)
        github.com/pingcap/ticdc@/cdc/owner.go:589 +0x7c
github.com/pingcap/ticdc/cdc.(*Server).handleChangefeedAdmin(0xc0000ac5a0, 0x247c660, 0xc0000b0a80, 0xc002f08d00)
        github.com/pingcap/ticdc@/cdc/http_handler.go:100 +0x1ba
net/http.HandlerFunc.ServeHTTP(0xc00034e8a0, 0x247c660, 0xc0000b0a80, 0xc002f08d00)
        net/http/server.go:2007 +0x44
net/http.(*ServeMux).ServeHTTP(0xc0000a2980, 0x247c660, 0xc0000b0a80, 0xc002f08d00)
        net/http/server.go:2387 +0x1bd
net/http.serverHandler.ServeHTTP(0xc0000ec000, 0x247c660, 0xc0000b0a80, 0xc002f08d00)
        net/http/server.go:2802 +0xa4
net/http.(*conn).serve(0xc000354780, 0x2484aa0, 0xc0007b7980)
        net/http/server.go:1890 +0x875
created by net/http.(*Server).Serve
        net/http/server.go:2927 +0x38e
```

owner node
```
[2020/05/08 11:55:37.657 +08:00] [INFO] [owner.go:418] ["handle admin job"] [changefeed=3f4c9771-46fd-4ed7-a6e5-31247d74f7bf] [type="resume changefeed"]
[2020/05/08 11:55:37.658 +08:00] [ERROR] [owner.go:526] ["owner exited with error"] [error="query detail id 3f4c9771-46fd-4ed7-a6e5-31247d74f7bf: changefeed not exists"] [errorVerbose="changefeed not exists\nquery detail id 3f4c9771-46fd-4ed7-a6e5-31247d74f7bf\ngithub.com/pingcap/ticdc/cdc/kv.CDCEtcdClient.GetChangeFeedInfo\n\tgithub.com/pingcap/ticdc@/cdc/kv/etcd.go:140\ngithub.com/pingcap/ticdc/cdc.(*Owner).handleAdminJob\n\tgithub.com/pingcap/ticdc@/cdc/owner.go:452\ngithub.com/pingcap/ticdc/cdc.(*Owner).run\n\tgithub.com/pingcap/ticdc@/cdc/owner.go:565\ngithub.com/pingcap/ticdc/cdc.(*Owner).Run\n\tgithub.com/pingcap/ticdc@/cdc/owner.go:523\ngithub.com/pingcap/ticdc/cdc.(*Server).run.func2\n\tgithub.com/pingcap/ticdc@/cdc/server.go:169\nruntime.goexit\n\truntime/asm_amd64.s:1357"]
```

4. Versions of the cluster

    - TiCDC version (execute `cdc version`):

```
cdc version
Release Version: v4.0.0-rc.1-12-g39e75c3
Git Commit Hash: 39e75c3524e2224715fb162aa84a380dc40cf2e6
Git Branch: master
UTC Build Time: 2020-05-08 03:58:42
Go Version: go version go1.13.3 linux/amd64
```

### PR DESCRIPTION
**Pull request by leoppro**
### What problem does this PR solve? 
fix #541 #542.(3)
ignore and warn the invalid admin jobs.

if users remove the changefeed, the owner will delete the changefeed info immediately and set 24h TTL to the changefeed status.

the CDC CLI will report a warning to tall users the changefeed has been deleted.

```
./cdc cli changefeed remove --changefeed-id 04dac05c-484d-49e2-8163-1cca0d7f2

./cdc cli changefeed query --changefeed-id 04dac05c-484d-49e2-8163-1cca0d7f2fb5
[2020/06/01 09:07:24.984 +00:00] [WARN] [client_changefeed.go:149] ["this changefeed has been deleted, the residual meta data will be completely deleted within 24 hours."]
{
  "info": null,
  "status": {
    "resolved-ts": 417071719969980417,
    "checkpoint-ts": 417071719707836417,
    "admin-job-type": 3
  },
  "count": 0,
  "task-status": []
}                                                             
```

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: cdc/owner.go [hunk 1]
```diff
@@ -475,10 +477,18 @@ func (o *Owner) handleAdminJob(ctx context.Context) error {
 			}
 		case model.AdminResume:
 			cfStatus, _, err := o.etcdClient.GetChangeFeedStatus(ctx, job.CfID)
+			if errors.Cause(err) != model.ErrChangeFeedNotExists {
```
Comment (leoppro) [inline cdc/owner.go:480]:
```suggestion
			if errors.Cause(err) == model.ErrChangeFeedNotExists {
```

File: cdc/owner.go [hunk 2]
```diff
@@ -475,10 +477,18 @@ func (o *Owner) handleAdminJob(ctx context.Context) error {
...
+				log.Warn("invalid admin job, changefeed not found", zap.String("changefeed", job.CfID))
+				continue
+			}
 			if err != nil {
 				return errors.Trace(err)
 			}
 			cfInfo, err := o.etcdClient.GetChangeFeedInfo(ctx, job.CfID)
+			if errors.Cause(err) != model.ErrChangeFeedNotExists {
```
Comment (leoppro) [inline cdc/owner.go:488]:
```suggestion
			if errors.Cause(err) == model.ErrChangeFeedNotExists {
```

File: cmd/client_changefeed.go [hunk 3]
```diff
@@ -143,6 +145,9 @@ func newQueryChangefeedCommand() *cobra.Command {
 				taskStatus = append(taskStatus, captureTaskStatus{CaptureID: captureID, TaskStatus: status})
 			}
 			meta := &cfMeta{Info: info, Status: status, Count: count, TaskStatus: taskStatus}
+			if info == nil {
+				log.Warn("this changefeed has been deleted, the residual meta data will be completely deleted within 24 hours.")
```
Comment (overvenus) [inline cmd/client_changefeed.go:149]: How do we test it if it is get deleted after 24 hours?

Comment (leoppro) [reply cmd/client_changefeed.go:149]: i will add a unit test for it

Comment (leoppro) [reply cmd/client_changefeed.go:149]: PTAL again

File: cdc/kv/etcd_test.go [hunk 4]
```diff
@@ -249,3 +249,26 @@ func (s *etcdSuite) TestPutAllChangeFeedStatus(c *check.C) {
...
+		ResolvedTs: 1,
+	})
+	err = s.client.SetChangeFeedStatusTTL(ctx, "test1", 2 /* seconds */)
+	c.Assert(err, check.IsNil)
+	status, _, err = s.client.GetChangeFeedStatus(ctx, "test1")
+	c.Assert(err, check.IsNil)
+	c.Assert(status, check.DeepEquals, &model.ChangeFeedStatus{
+		ResolvedTs: 1,
+	})
+	time.Sleep(3 * time.Second)
```
Comment (overvenus) [inline cdc/kv/etcd_test.go:271]: Could you speed up the test? Sleeping 3s have noticeable delay when running test. 😅

File: cdc/kv/etcd_test.go [hunk 5]
```diff
@@ -249,3 +249,26 @@ func (s *etcdSuite) TestPutAllChangeFeedStatus(c *check.C) {
...
+	err := s.client.PutChangeFeedStatus(ctx, "test1", &model.ChangeFeedStatus{
+		ResolvedTs: 1,
+	})
+	c.Assert(err, check.IsNil)
+	status, _, err := s.client.GetChangeFeedStatus(ctx, "test1")
+	c.Assert(err, check.IsNil)
+	c.Assert(status, check.DeepEquals, &model.ChangeFeedStatus{
+		ResolvedTs: 1,
+	})
+	err = s.client.SetChangeFeedStatusTTL(ctx, "test1", 2 /* seconds */)
```
Comment (leoppro) [inline cdc/kv/etcd_test.go:264]:
```suggestion
	err = s.client.SetChangeFeedStatusTTL(ctx, "test1", 1 /* seconds */)
```

File: cdc/kv/etcd_test.go (hunk 4 above)
Comment (leoppro) [inline cdc/kv/etcd_test.go:271]:
```suggestion
	time.Sleep(1.5 * time.Second)
```

**Pull request merged**